out_tcp2/tcp2.c

```diff
@@ -0,0 +1,126 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019      The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_output.h>
+#include <fluent-bit/flb_pack.h>
+#include "tcp2.h"
+
+static int cb_tcp_init(struct flb_output_instance *ins,
+                          struct flb_config *config, void *data)
+{
+    (void) data;
+    int ret;
+    int io_flags = 0;
+    const char *tmp;
+    struct flb_upstream *upstream;
+    struct flb_out_tcp2 *ctx = NULL;
+
+     /* Allocate plugin context */
+    ctx = flb_calloc(1, sizeof(struct flb_out_tcp2));
+    if(!ctx) {
+        return 0;
+    }
+    ctx->ins = ins;
+
+    /* Set default network configuration if not set */
+    flb_output_net_default("127.0.0.1", 5170, ins);
+
+    io_flags = FLB_IO_TCP;
+
+     /* Upstream context */
+    upstream = flb_upstream_create(config,
+                                   ins->host.name,
+                                   ins->host.port,
+                                   io_flags, (void *) &ins->tls);
+    if (!upstream) {
+        printf("Could not create upstream context");
+        flb_free(ctx);
+        return 0;
+    }
+    
+    /* Set different parameters */
+    ctx->out_format = FLB_PACK_JSON_FORMAT_NONE;
```

Comment:
please make sure that the example plugin writes the data in JSON format instead of msgpack, so an end user can see the data when testing.